dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java

```diff
@@ -725,28 +725,32 @@ public VersionTypeValidation validateWorkflowSet(Set<SourceFile> sourcefiles, St
         Optional<SourceFile> mainDescriptor = filteredSourcefiles.stream().filter((sourceFile -> Objects.equals(sourceFile.getPath(), primaryDescriptorFilePath))).findFirst();
 
         boolean isValid = true;
-        String validationMessage = null;
+        StringBuilder validationMessage = new StringBuilder();
         Map<String, String> validationMessageObject = new HashMap<>();
 
         if (mainDescriptor.isPresent()) {
             Yaml yaml = new Yaml();
             String content = mainDescriptor.get().getContent();
             if (content == null || content.isEmpty()) {
                 isValid = false;
-                validationMessage = "Primary descriptor is empty.";
+                validationMessage.append("Primary descriptor is empty.");
             } else if (!content.contains("class: Workflow")) {
                 isValid = false;
-                validationMessage = "Requires class: Workflow.";
+                validationMessage.append("A CWL workflow requires 'class: Workflow'.");
+                if (content.contains("class: CommandLineTool") || content.contains("class: ExpressionTool")) {
+                    String cwlClass = content.contains("class: CommandLineTool") ? "CommandLineTool" : "ExpressionTool";
```

Comment:
This is nit-picky and you don't need to change the PR for it, but FYI string searches can be relatively expensive so you should generally try to avoid searching for the same substring more than once. Imagine if `content` was 100K or bigger -- worst case it will go through the whole string two times if there's not a match for CommandLineTool, plus twice more for however far into the string the match for ExpressionTool is. Unless the compiler/runtime does some fancy optimization, but I wouldn't rely on it.

OTOH, descriptors aren't typically that big, it will still be fast, and this code is not frequently executed.